### ISSUE TITLE
Some issues to clarify

### ISSUE THREAD

**Issue by mhmtsarigul** [facebookresearch/synsin#32]
Hello. Thanks for great work.  

I have some questions about the project. 

In realestate10k.py, you calculate the K matrix with the intrinsics of the camera but then you use the self.K which is an eye matrix defined in __init__ to return in loader. In other datasets while intrinsics are same for all the dataset you define the K in the __init__ and use it in getitem which is clearly correct for me. But in realestate why you don't use the correct K matrix.

`
            origK = np.array(
                [
                    [intrinsics[0], 0, intrinsics[2]],
                    [0, intrinsics[1], intrinsics[3]],
                    [0, 0, 1],
                ],
                dtype=np.float32,
            )
            K = np.matmul(self.offset, origK)

            origP = extrinsics.reshape(3, 4)
            P = np.matmul(K, origP)  # Merge these together to match habitat
            P = np.vstack((P, np.zeros((1, 4), dtype=np.float32))).astype(
                np.float32
            )
            P[3, 3] = 1

            Pinv = np.linalg.inv(P)

            cameras += [
                {
                    "P": P,
                    "Pinv": Pinv,
                    "OrigP": origP,
                    "K": self.K,
                    "Kinv": self.invK,
                }
            ]

        return {"images": rgbs, "cameras": cameras}`

And one another issue, in kitti.py there is a matrix multiplication which i don't understand. Multiplying offset with K will make the [0,0] and [1,1] cells' values 2*(718.9/256) and will make [0,2],[1,2] cells, 0. And -1 does not multiply with anything just zero. Which point do I miss? Can you clarify?  

`        K = np.array(
                [718.9 / 256., 0., 128 / 256., 0, \
                 0., 718.9 / 256., 128 / 256., 0, \
                 0., 0., 1., 0., \
                 0., 0., 0., 1.]).reshape((4, 4)).astype(np.float32) 
        
        offset = np.array(
            [[2, 0, -1, 0], [0, -2, 1, 0], [0, 0, 1, 0], [0, 0, 0, 1]],  # Flip ys to match habitat
            dtype=np.float32,
        )  # Make z negative to match habitat (which assumes a negative z)

        K = np.matmul(offset, K)
`

**Comment by DRealArun** [comment]
@mhmtsarigul: Thanks for bringing this up. I have been facing a similar conundrum.

**Comment by oawiles** [comment]
For the K, we matrix multiply it with the extrinsics. It's been a while but I think it then matched the other datasets where the K is fixed, which is an assumption we have in the forward pass.

I'm not sure about the second question (as to what the question is). Basically, there are implicit assumptions in the code as to how to interpret the X/Y axes which may not match the cameras in the dataset. As a result, I had to flip the Ys (as the offset is doing) and actually in this case make the Zs negative in order to match those assumptions.

**Comment by DRealArun** [comment]
@oawiles: Sorry for the delay, according to my understanding, the offset matrix, if it is supposed to just flip the axis, shouldn't it just resemble a product of few rotation matrices? I am not able to understand how that offset matrix corresponds to the axis flips that you mentioned. Another reason for my confusion is that the offset matrix is elementwise multiplied to the K matrix.

**Comment by oawiles** [comment]
Yes you could do it that way. I did not modularise the code (which may have been clearer) but did it this way. I use np.matmul -- this is matrix multiplication? Not elementwise. It's moving the x/y values from a range of [0, 1] --> [-1, 1] in addition to flipping the y axes. I think that's everything it's doing (but don't quite remember).
